### PR TITLE
[Kernels]Update triton kernels into 2.1.0

### PR DESCRIPTION
**Pull request by tiandiao123**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

This PR is used to further to update triton kernels into 2.1.0, and also reconstruct triton context-attention kernels to be compatible with triton 2.0.0 and triton 2.1.0. 
Also, added a build script to install relate packages. 

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -16,3 +16,4 @@ ray
 sentencepiece
 google
 protobuf
+pytest
```
Comment (ver217) [inline requirements/requirements.txt:19]: This should be test requirements, right?

Comment (ver217) [reply requirements/requirements.txt:19]: Do we need to update triton's version in requirements-test.txt?

Comment (tiandiao123) [reply requirements/requirements.txt:19]: > This should be test requirements, right?
right, will fix it.

Comment (tiandiao123) [reply requirements/requirements.txt]: fix it.

File: examples/inference/colossal_llama2_test.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+@pytest.mark.dist
+@rerun_if_address_is_in_use()
+@clear_cache_before_run()
+def test_llama(args):
+    spawn(check_llama, args.tp_size, args=args)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--path", type=str, default = "hpcai-tech/Colossal-LLaMA-2-7b-base", help="Model path")
```
Comment (yuehuayingxueluo) [inline examples/inference/colossal_llama2_test.py:75]: Will directly using the path of the official website cause CI failure?

Comment (tiandiao123) [reply examples/inference/colossal_llama2_test.py:75]: it is not a test, it is just a demo for testing colossal-llama2. CI will not run this.

**Review by yuehuayingxueluo — APPROVED** [review]

File: examples/inference/colossal_llama2_test.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            print(output_text)
+
+
+def check_llama(rank, world_size, port, args):
+    disable_existing_loggers()
+    colossalai.launch(config={}, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
+    run_llama_test(args=args)
+
+
+@pytest.mark.skipif(not CUDA_SUPPORT, reason="kv-cache manager engine requires cuda version to be higher than 11.5")
```
Comment (ver217) [inline examples/inference/colossal_llama2_test.py:65]: Test shoud not be in `examples/` folder. What's exactly this file for?

File: examples/inference/colossal_llama2_test.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import os
```
Comment (CjhHa1) [inline examples/inference/colossal_llama2_test.py:1]: may need move this to 'test' folder and change a name

Comment (tiandiao123) [reply examples/inference/colossal_llama2_test.py:1]: OK!

File: examples/inference/colossal_llama2_test.py (hunk 3 above)
Comment (tiandiao123) [reply examples/inference/colossal_llama2_test.py:65]: just for a colossal-llama2 demo. Originally, we don't have colossal-llama2 demo

File: examples/inference/colossal_llama2_test.py (hunk 4 above)
Comment (tiandiao123) [reply examples/inference/colossal_llama2_test.py]: fixed it.

File: examples/inference/colossal_llama2_test.py (hunk 3 above)
Comment (tiandiao123) [reply examples/inference/colossal_llama2_test.py]: fixed it.

**Review by CjhHa1 — APPROVED** [review]

**Review by ver217 — APPROVED** [review]

**Pull request merged**